src/main/scala/uk/gov/nationalarchives/aggregate/processing/AggregateProcessingLambda.scala

```diff
@@ -101,8 +101,12 @@ class AggregateProcessingLambda extends RequestHandler[SQSEvent, Unit] {
           IO(Nil)
         } else {
           val clientSideMetadataInput = assetProcessingResults.flatMap(_.clientSideMetadataInput)
-          val input = AddFileAndMetadataInput(consignmentId, clientSideMetadataInput, None, Some(userId))
-          persistenceApi.addClientSideMetadata(input)
+          val addFileAndMetadataInput = AddFileAndMetadataInput(consignmentId, clientSideMetadataInput, None, Some(userId))
+          val updateParentFolderInput = UpdateParentFolderInput(consignmentId, parentFolder = clientSideMetadataInput.head.originalPath.split("/").head, Some(userId))
+          for {
+            _ <- persistenceApi.addClientSideMetadata(addFileAndMetadataInput)
+            _ <- persistenceApi.addParentFolder(updateParentFolderInput)
```

Comment:
I'd put the parent folder request first, as it happens before the metadata input in the current frontend journey